### ISSUE TITLE
How to embed an image

### ISSUE THREAD

**Issue by joshcomley** [alonrbar/easy-template-x#4]
I can see that you support binary, but how exactly does one go about embedding an image from the JSON data into the template?

Many thanks!

**Comment by alonrbar** [comment]
Hi,

Embedding an image is currently not supported. 

If want you are welcome to add the necessary plugin. The general idea would be to:

1. Insert the correct xml into the document (see the Office Open XML spec [here](http://officeopenxml.com/drwPic.php) and line 605 from docx-templates [here](https://github.com/guigrpa/docx-templates/blob/master/packages/docx-templates/src/processTemplate.js))
2. Add the image to the docx zip package.
3. Add a .rel entry of the image (Relationships section [here](http://officeopenxml.com/anatomyofOOXML.php) and line 239 [here](https://github.com/guigrpa/docx-templates/blob/4c007ad5b5f436996de679d6a398a425f07d5f41/packages/docx-templates/src/mainBrowser.js)).

**Comment by jjhbw** [comment]
First of all, thanks for this library! I found that it hits the sweet spot for typescript users.

For those stumbling on this issue, maybe the following quick-and-dirty solution to image templating can help.

1. Add a placeholder image to your docx template file.
2. Unzip the docx template and take note of the location of the image in the unzipped file tree.

Then you can code something like this. Replace `target_path` with the location of the placeholder.
```typescript
import { TemplateHandler } from 'easy-template-x'
import JSZip from 'jszip'
import { saveAs } from 'file-saver'

export async function renderDocxTemplate() {
    const template_file = await (await fetch('/report_template.docx')).blob()
    const target_image = await (await fetch('/test_img.png')).blob()

    const data = {
        text_template: 'new phone who dis'
    }

    const handler = new TemplateHandler()
    const only_text_rendered = await handler.process(template_file, data)

    // // Replace the placeholder images
    // Read the produced binary zip file again to replace the placeholder images
    const new_zip = new JSZip()
    await new_zip.loadAsync(only_text_rendered)

    // Replace the image
    const target_path = 'word/media/image1.png'
    new_zip.remove(target_path)
    new_zip.file(target_path, target_image, { binary: true })

    // Produce an output buffer
    const blob = await new_zip.generateAsync({ type: "blob" })

    saveAs(blob, 'report.docx')
}
```

**Comment by alonrbar** [comment]
Glad you like it, and a nice workaround too, thanks!
Though you might want to know that while it might take a while I am working on a proper solution for this on the `develop` branch.

**Comment by alonrbar** [comment]
The **ImagePlugin** is now fully operational and available in the develop branch 🎉

Since there are some breaking changes (checkout the [changelog](https://github.com/alonrbar/easy-template-x/blob/develop/CHANGELOG.md)) I'll only release a new version once I'll finish updating the readme file. That probably won't happen today but, in the meanwhile you can reference the develop branch directly at this commit **0ac74019** by using the technique described here: https://stackoverflow.com/questions/14187956/npm-install-from-git-in-a-specific-version#27630247

**Comment by alonrbar** [comment]
Released v0.6.0 which includes the ImagePlugin.

**Comment by rushabhshah95** [comment]
@alonrbar  - This embed image plugin also works for imageUrl? I am bit confused to implement that. could you please help me with it?
